config/routes.rb

```diff
@@ -1,4 +1,8 @@
 Rails.application.routes.draw do
+  root to: "users#index"
+  get "users/:author_id/posts", to: "posts#index", as: "user_posts"
+  get "users/:author_id/posts/:id", to: "posts#show", as: "user_post"
+  get "users/:id", to: "users#show", as: "user"
```

Comment:
- [ ] [optional] @aamirkhan2478 great work on your routes, however, I would recommend using resources to create your routes its easier to scale your routes in the future and makes your code cleaner below is an example: 
 ```suggestion
 resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end

```